shared/oae/api/oae.api.util.js

```diff
@@ -1132,6 +1184,147 @@ define(['exports', 'require', 'jquery', 'underscore', 'oae.api.config', 'markdow
                 // Add a label to the autosuggest input field for accessibility
                 $('.as-input', $list).before('<label class="sr-only" for="' + $('.as-input', $list).attr('id') + '">' + options.startText + '</label>');
 
+                /**
+                 * Parse the current auto-suggest input into comma or space-separated tokens
+                 *
+                 * @return {String[]}  An array of all user-inputted tokens
+                 * @api private
+                 */
+                var getTokens = function() {
+                    return $element.val().split(/[\s,]+/);
+                };
+
+                /**
+                 * Insert the specified email address as invitation entries in the auto-suggest
+                 * field
+                 *
+                 * @param  {String[]}   emailAddresses  The email addresses to embed into the field
+                 * @api private
+                 */
+                var insertEmails = function(emailAddresses) {
+                    _.each(emailAddresses, function(emailAddress) {
+                        // Render the template for the email invitation entry
+                        var $li = $(template().render($('#autosuggest-email-template', $autosuggestTemplates), {
+                            'emailAddress': emailAddress
+                        }).trim());
+
+                        // Insert the template onto the DOM
+                        $element.parent('li.as-original').before($li);
+
+                        // Attach our data to the entry
+                        $li.data({
+                            'originalData': {
+                                'resourceType': 'email',
+                                'id': emailAddress,
+                                'dipslayName': emailAddress
+                            }
+                        });
+
+                        // Bind the remove event to the entry
+                        // TODO: Shouldn't there just be something like a $('a.as-close', $ul) to handle this for all the li's?
```

Comment:
Don't we or the autosuggest plugin have a function that inserts a new element into the auto suggest and takes care of doing things like binding the remove button?